test/es-module/test-esm-detect-ambiguous.mjs

```diff
@@ -246,9 +246,12 @@ describe('Module syntax detection', { concurrency: !process.env.TEST_PARALLEL },
         'await Promise.resolve(); console.log("executed");',
       ]);
 
-      strictEqual(stderr, '');
-      strictEqual(stdout, 'executed\n');
-      strictEqual(code, 0);
+      match(
+        stderr,
+        /Top-level await is not supported in CommonJS modules\. To use top-level await, add "type": "module" to your package\.json or rename the file to use the \.mjs extension/
```

Comment:
This test is titled “permits top-level `await`”. This PR shouldn’t have the effect of prohibiting top-level `await` in ambiguous files that get detected to be ESM. This isn’t a test that should need updating as part of the goals of this PR.